packages/flutter/lib/src/services/fetch.dart

```diff
@@ -39,7 +41,10 @@ Future<UrlResponse> fetch(UrlRequest request) async {
   UrlLoaderProxy loader = new UrlLoaderProxy.unbound();
   try {
     _networkService.ptr.createUrlLoader(loader);
+    ShutdownListener shutdownListener = () => loader.close();
+    lifecycle.addShutdownListener(shutdownListener);
     UrlResponse response = (await loader.ptr.start(request)).response;
+    lifecycle.removeShutdownListener(shutdownListener);
```

Comment:
In principle, wouldn't need need to do this for every await call on every mojo handle?  That seems crazy.